python/cudf/dataframe/dataframe.py

```diff
@@ -1312,10 +1314,11 @@ def transpose(self):
     def T(self):
         return self.transpose()
 
-    def merge(self, right, on=None, how='left', lsuffix='_x', rsuffix='_y',
+    def merge(self, right, on=None, how='inner', left_on=None, right_on=None,
+              left_index=False, right_index=False, lsuffix='_x', rsuffix='_y',
               type="", method='hash'):
-        """Merge GPU DataFrame objects by performing a database-style join operation
-        by columns or indexes.
+        """Merge GPU DataFrame objects by performing a database-style join
+        operation by columns or indexes.
```

Comment:
There is some value to keeping the first line of a docstring to a single line.  These often end up coming up in tables of API documentation, like in http://docs.dask.org/en/latest/dataframe-api.html .  Keeping them to a single line, followed by a blank line, followed by a longer paragraph if necessary is a common standard.  See http://docs.dask.org/en/latest/develop.html#docstrings

(Though of course, cudf may use different standards.  Please disregard if desired)

Reply:
Hmm. This contests with Flake8. I'm more than happy to have long lines but Flake8 produces a warning and it's likely to fail style integration tests. I'll check on that as well.

Reply:
Yup, you would have to have a shorter title line.  Typically they're under 80 characters.  To be clear, the ask here isn't "put all of this on one line" it's "make this short enough so that it fits on one line".

Again though, I'm applying style conventions from outside cudf to cudf, which may not be what we want to do.

Reply:
@mrocklin We likely violate this in a number of places elsewhere where lets keep this for now and revamp the documentation in general in 0.7+.